runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1186,12 +1851,29 @@ TR_VectorAPIExpansion::vectorizeLoadOrStore(TR_VectorAPIExpansion *opt, TR::Node
          traceMsg(comp, "   created new vector symRef #%d for #%d\n", vecSymRef->getReferenceNumber(), symRef->getReferenceNumber());
 
       }
+
+   TR::ILOpCodes opcode;
+
    if (node->getOpCode().isStore())
-      TR::Node::recreate(node, TR::ILOpCode::createVectorOpCode(opCodeType.isVector() ? TR::vstore : TR::mstore, opCodeType));
+      opcode = TR::ILOpCode::createVectorOpCode(opCodeType.isVector() ? TR::vstore : TR::mstore, opCodeType);
+   else
+      opcode = TR::ILOpCode::createVectorOpCode(opCodeType.isVector() ? TR::vload : TR::mload, opCodeType);
+
+   if (!newLoad)
+      {
+      TR::Node::recreate(node, opcode);
+      }
    else
-      TR::Node::recreate(node, TR::ILOpCode::createVectorOpCode(opCodeType.isVector() ? TR::vload : TR::mload, opCodeType));
+      {
+      TR_ASSERT_FATAL(!node->getOpCode().isStore(), "Should be a load node");
+      node = TR::Node::create(node, opcode, 0);
```

Comment:
You assign the return value of `TR::Node::recreate()` to `node` here, while you don't do that at L.1864.  Is that OK?
I searched for the definition of `TR::Node::recreate()` with three arguments, but I couldn't find it.

Reply:
This is by design. In one case, we modify the node in place and in another (new) case we create a new node.